### PR TITLE
Nischal

### PR DESCRIPTION
**Pull request by ngprnk**
Added method RenderURltopdf in Pupetter 
added envorinment variable to check the mode if  env file doesnt exists then defaults to development 
added a new route get-pdf so that it doesnot effect ch40 
logs full error  if it application is running on a development mode

### CODE REVIEW

File: src/config/config.js [hunk 1]
```diff
@@ -8,4 +11,8 @@ export default {
     errorFile: 'logs/error.log',
     logFile: 'logs/combined.log'
   },
+
+  mode: {
+    environment: process.env.NODE_ENVIRONMENT || 'development'
```
Comment (matthaywardwebdesign) [inline src/config/config.js]: Looks good here. Are we able to remove that extra space on line 14 and consider using `process.env.NODE_ENV` as it is the standard way to set the node environment.

File: src/controllers/raw/rawHTMLToPDF.js [hunk 2]
```diff
@@ -4,17 +4,23 @@ import uuid from 'uuid/v4';
 
 export default async ( req, res ) => {
   /* Get the html body */
-  const { html, sendFile } = req.body;
+  const { html, sendFile, urlToPDF } = req.body;
```
Comment (matthaywardwebdesign) [inline src/controllers/raw/rawHTMLToPDF.js]: I think just `url` would be a better field name. Thoughts?

File: src/config/config.js (hunk 1 above)
Comment (ngprnk) [reply src/config/config.js]: ok will do that

File: src/lib/api/ErrorHandler.js [hunk 3]
```diff
@@ -2,7 +2,8 @@
 import os from 'os';
 import config from 'config';
 import { DEFAULT_ERROR_MESSAGE } from 'constants';
-
+import Logger from '../logging';
+Logger.warn("Dev Mode: "+ config.mode.environment+ '\n' );
```
Comment (matthaywardwebdesign) [inline src/lib/api/ErrorHandler.js]: Maybe just only log the environment if we are in a non production environment.

File: src/lib/api/ErrorHandler.js [hunk 4]
```diff
@@ -55,6 +56,11 @@ const errorHandler = ( err, req, res, next ) => {
    * as well as including hostname and version
    */
   res.status( code );
+  /* if the mode is development send the full error in the console  for debuggig purposes*/
```
Comment (matthaywardwebdesign) [inline src/lib/api/ErrorHandler.js]: Spelling mistake here :)

File: src/lib/api/ErrorHandler.js [hunk 5]
```diff
@@ -55,6 +56,11 @@ const errorHandler = ( err, req, res, next ) => {
...
+  if( config.mode.environment === 'development' ){
+    console.log( err);
```
Comment (matthaywardwebdesign) [inline src/lib/api/ErrorHandler.js]: Let's use the proper logger here :)

File: src/lib/puppeteer/Puppeteer.js [hunk 6]
```diff
@@ -24,6 +24,25 @@ class Puppeteer {
     await page.goto( `http://localhost:${config.api.port}/docs/${filename}.html`, { waitUntil: 'networkidle2' });
     await page.pdf({ path: `tmp/${filename}.pdf`, ...combinedOptions });
   }
+
+  async renderURLToPDF ( filename, pdfOptions={}, otherOptions={}){
```
Comment (matthaywardwebdesign) [inline src/lib/puppeteer/Puppeteer.js]: Nice work on this function! Ideally I'd like to combine the options field into a single object. Maybe a better method prototype would be `renderURLToPDF( filename, data, options )`. Thoughts?

File: src/lib/puppeteer/Puppeteer.js [hunk 7]
```diff
@@ -24,6 +24,25 @@ class Puppeteer {
...
+
+    /* Load the browser if it isn't already loaded */
+    if ( !this.browser ) {
+      await this.loadBrowser();
+    }
+
+    /* Create a new blank page */
+    const page = await this.browser.newPage();
+
+    /* Navigate to the html */
```
Comment (matthaywardwebdesign) [inline src/lib/puppeteer/Puppeteer.js:44]: Let's just fix up this comment to reflect actually functionality.

**Comment by matthaywardwebdesign** [comment]
Great work on this one! I've added a bunch of little comments :)

File: src/config/config.js (hunk 1 above)
Comment (matthaywardwebdesign) [reply src/config/config.js]: Thanks!

File: src/controllers/raw/rawHTMLToPDF.js (hunk 2 above)
Comment (ngprnk) [reply src/controllers/raw/rawHTMLToPDF.js]: ok will update that

**Comment by matthaywardwebdesign** [comment]
One more thing - If you get a chance would you please be able to add some docs on how to use the new URL to PDF functionality.

File: src/lib/api/ErrorHandler.js (hunk 3 above)
Comment (ngprnk) [reply src/lib/api/ErrorHandler.js]: Done

**Pull request merged**